dss_datamover/socket_communication.py

```diff
@@ -138,67 +139,62 @@ def recv_json(self, format="JSON", timeout=RECV_TIMEOUT):
         :timeout: default 60 seconds
         :return: Return received data in json format.
         """
-        msg_len = None
+        self.socket.settimeout(timeout)
         msg = "{}"
+        time_started = datetime.now()
 
         try:
             msg_len_in_bytes = b''
             received_msg_len_size = 0
-            time_started = datetime.now()
-            # Iterate till we receive 10 bytes or timeout.
-            while received_msg_len_size < MESSAGE_LENGTH:
-                received_msg_len_in_bytes = self.socket.recv(MESSAGE_LENGTH - received_msg_len_size)
-                received_msg_len_size += len(received_msg_len_in_bytes)
-                msg_len_in_bytes += received_msg_len_in_bytes
-                time_spent_in_seconds = (datetime.now() - time_started).seconds
-                if time_spent_in_seconds >= timeout:
-                    raise socket.timeout("ClientSocket: Timeout ({} seconds) from recv function".format(
-                        time_spent_in_seconds))
-            if msg_len_in_bytes != b'':
-                msg_len = int(msg_len_in_bytes)
-        except socket.timeout as e:
-            raise e
-        except socket.error as e:
-            raise socket.error("ClientSocket: Incorrect message length - {}".format(e))
-        except ValueError as e:
-            raise ValueError("ClientSocket: ValueError - {}".format(e))
 
-        if msg_len:
+            # first receive message length from payload
+            msg_len_in_bytes = self.socket.recv(MESSAGE_LENGTH)
+            msg_len = int(msg_len_in_bytes)
+
+            # now retrieve rest of payload from buffer based on msg_len
+            # TODO: handle rare edge case where only partial bytes were read before socket termination
             msg_body = b''
-            # Iterate till we receive desired number of bytes.
             received_data_size = 0
             while received_data_size < msg_len:
                 data_size = msg_len - received_data_size
-                try:
-                    received_data = self.socket.recv(data_size)
-                    received_data_size += len(received_data)
-                    msg_body += received_data
-                except socket.timeout as e:
-                    self.logger.error("ClientSocket: Timeout-{}".format(e))
-                    break
-                except socket.error as e:
-                    self.logger.error("ClientSocket: {}".format(e))
+                received_data = self.socket.recv(data_size)
+                received_data_size += len(received_data)
+                msg_body += received_data
 
+            # process received payload
             if msg_body == b'':
                 raise RuntimeError("ClientSocket: Empty message for message length -{}".format(msg_len))
-
-            if msg_body:
+            elif msg_body:
                 if len(msg_body) == msg_len:
```

Comment:
why do we need to do len(msg_body) when the received_data_size is available on the top.